src/DefaultEventLogger.ts

```diff
@@ -54,6 +55,37 @@ class DefaultEventLogger implements EventLogger, EventPreProcessor, EventPostPro
       return result
     }
 
+    createNewTraceMetadata(service: string, sampled?: number | undefined, flags?: number | undefined, timestamp?: string | undefined): EventTraceMetadata {
+
+      let newMeta = new EventTraceMetadata(service, new Uuid(), new Uuid(), undefined, sampled, flags, timestamp);
+      return newMeta;
+    }
+
+    createChildTraceMetadata(parentTraceMetadata: EventTraceMetadata): EventTraceMetadata {
+      return new EventTraceMetadata(parentTraceMetadata.service, parentTraceMetadata.traceId, new Uuid(), parentTraceMetadata.spanId)
```

Comment:
Any reason that additional params are not set? 
e.g:
- `flags`
- `service` will need to be configurable <-- this is the name of the span that will be displayed on the trace.
- `timestamp` will also need to be set

Reply:
Since they're optional they could be set afterward on the returned value.
But I can add them as optional parameters

Reply:
DONE. `6.4.6-snapshot`